stdlib/LinearAlgebra/src/structuredbroadcast.jl

```diff
@@ -9,53 +9,60 @@ StructuredMatrixStyle{T}(::Val{2}) where {T} = StructuredMatrixStyle{T}()
 StructuredMatrixStyle{T}(::Val{N}) where {T,N} = Broadcast.DefaultArrayStyle{N}()
 
 const StructuredMatrix = Union{Diagonal,Bidiagonal,SymTridiagonal,Tridiagonal,LowerTriangular,UnitLowerTriangular,UpperTriangular,UnitUpperTriangular}
-Broadcast.BroadcastStyle(::Type{T}) where {T<:StructuredMatrix} = StructuredMatrixStyle{T}()
+for ST in Base.uniontypes(StructuredMatrix)
+    @eval Broadcast.BroadcastStyle(::Type{<:$ST}) = StructuredMatrixStyle{$ST}()
```

Comment:
```suggestion
    @eval Broadcast.BroadcastStyle(::Type{<:$ST}) = $(StructuredMatrixStyle{ST}())
```
If we are going to use `eval` on this we might as well splice in the exact result for it